docs/pallas/tpu/core_map.md

```diff
@@ -0,0 +1,358 @@
+# Pallas Core-specifc Programming
+
+In this guide, we explore using `pl.core_map` to write Pallas kernels. Compared with `pallas_call`, `core_map` offers a few key characteristics:
+
+* **Per-core level programming**: You write code for an TPU/GPU core, not for a JAX device. This is crucial if you want to specifically control a core, or how cores communicate and distribute work among one another.
+
+* **Flexible pipelining**: You have the option to write pipelining communications on your own, instead of relying on Pallas grids and specs. This is helpful if your pipeline diverges from the standard "copy-in, compute & copy-out" pattern.
+
+* **Collectives**: Since `core_map` allows inter-core communications, it is especially helpful when writing collectives on the core level.
```

Comment:
I thought pallas_call allows this too?

Reply:
Yes, I mentioned that in the pipelining section. `pallas_call` automatically distribute works along the `PARALLEL` axis, only for pipelining, so it's a more restricted programming model than `core_map`.